Documentation/filesystem-crd.md

```diff
@@ -0,0 +1,67 @@
+---
+title: File System CRD
+weight: 20
+indent: true
+---
+
+# Creating Shared File Systems
+
+Rook allows creation and customization of shared file systems through the custom resource definitions (CRDs). The following settings are available
+for file systems.
+
+## Sample
+
+```yaml
+apiVersion: rook.io/v1alpha1
+kind: Filesystem
+metadata:
+  name: myfs
+  namespace: rook
+spec:
+  metadataPool:
+    replicated:
+      size: 3
+  dataPools:
+    - erasureCoded:
+       dataChunks: 6
+       codingChunks: 2
+  metadataServer:
+    activeCount: 2
```

Comment:
to start with we should make them user-configurable until we understand the patterns better and can fully automate.

Reply:
I actually come from the other spectrum. I would always advocate to not expose configs to the users if possible. As a user i get easily overwhelmed when i see a lot of dials and configs. I prefer simplicity, easy to use and easy to understand. I would expose only configs if the users ask/requires.

Reply:
They can just go with the defaults in the yaml and it will be fine. These are two options where clearly an admin will want to reconfigure in some scenarios so it doesn't seem appropriate to hide the options completely. There are plenty of other options that are hidden and the user will have to use the ceph tools to configure advanced options, but these are simple and the defaults are sufficient for 90% of cases.